### ISSUE TITLE
Add support for StopInfo project to help blind riders to better locate bus stops

### ISSUE THREAD

**Issue by barbeau** [OneBusAway/onebusaway-android#103]
More info about the StopInfo project on OBA Dev list [here](https://groups.google.com/forum/#!topic/onebusaway-developers/PKHS8MtAuO0).

PR that added this feature to OBA iPhone:
https://github.com/OneBusAway/onebusaway-iphone/pull/245

To implement this, we'd need to:

1. Consume StopInfo server url from Regions API
2. Show info button on arrival info fragment, if the StopInfo url is not null for the current region
3. Direct the user to the StopInfo server url when they tap on the info button

More details are in the above links.

**Comment by barbeau** [comment]
It's been a while since I updated this issue - here's what the stop info icon currently looks like in the app:
![image](https://cloud.githubusercontent.com/assets/928045/10232742/63046c48-6858-11e5-9b82-2f3a29a64094.png)

**Comment by caitbonnar** [comment]
@barbeau Thanks for the update! It looks good -- but is it possible to make it a tad bigger? That's mostly for people with low vision who don't use TalkBack. Which brings up another point-- is the button accessible via TalkBack? If so, what does the label read? Is there a hint?

**Comment by barbeau** [comment]
@caitbonnar Was just about to ping you with some questions re: this :).  Yes, I can make the image itself bigger, although much larger it starts to really dominate the area.  The touch area (see dark box surrounding icon below) is larger than the icon image, and I can potentially increase that more also:

![image](https://cloud.githubusercontent.com/assets/928045/10233210/2d0e077c-685b-11e5-8579-2f27fdb03b07.png)

Content description for Talkback is "About this stop".

Another option would be to move it (or supplement it) with an entry in the overflow menu that you get to by tapping on the three dots in the Action Bar:

![image](https://cloud.githubusercontent.com/assets/928045/10233360/fe3e74ee-685b-11e5-9aea-b6316ac28147.png)

... with a name like "Stop Info" or "Stop Details", again with Talkback description too.  I was wondering if this fixed position in the overflow menu might actually be more accessible?

The issue with the info button in the header is that it moves around depending on the length of the stop name.  But I figured you'd have the best idea of how either option would work for BLV users.

**Comment by caitbonnar** [comment]
@barbeau Thanks for being open to the feedback! :) 
I'm not very familiar with TalkBack, but if it is anything like VoiceOver, there will be a way to navigate through UI elements without having to physically locate them on the screen (i.e. you can swipe right or left and the VO cursor will focus on elements deemed as accessibility elements in some fixed order -- in VO this is generally top left to bottom right of the screen in left-to-right languages). So it's okay that the stop name length is variable as long as that button is accessible. 

That being said, I do like having it be a menu option (something I've pushed for in OBA iOS), because I believe fixed, ordered lists (as long as they're relatively short) are easy to traverse with a screenreader. But that's based on my intuition more than it is research. 

I can ask some B/LV friends their opinion too, if you'd like!

**Comment by barbeau** [comment]
@caitbonnar Ok, thanks.  Yes, getting feedback from B/LV contacts would be great.  It would also be great if any of them are willing to sign up for alpha updates, which includes the above stop info button:
https://github.com/OneBusAway/onebusaway-android/blob/master/BETA_TESTING.md#register-as-an-alpha-tester

I'd like to get their feedback not only on the stop info button, but the new UI in general.  Lots of things have changed from the current public version, and if we broke anything that B/LV users depend on it would be good to know now rather than later.

On a related note - if we can get the `stop_code` added to the StopInfo page (see https://github.com/OneBusAway/onebusaway-stopinfo/issues/1), then I'll likely show this info button and/or menu option in all regions.  In regions with a StopInfo server, it will direct to the StopInfo page.  In regions without a StopInfo Server, it will show a dialog with the stop details from GTFS, including `stop_code` - we've had ongoing requests to display the `stop_code` in some way so users know the number to report to the agency if there are problems (https://github.com/OneBusAway/onebusaway-android/issues/171).

**Comment by caitbonnar** [comment]
@barbeau Great, I'll send a request out! I know of at least one blind OBA Android user who's been wanting to give accessibility feedback for awhile -- sounds like a good time to do that!

I'll ask @alanborning about possibly getting the stop code attached as well. Neither @meganca nor I are at UW anymore, so I'm not sure StopInfo is being maintained by anyone directly anymore. It might be a good time to make it open source as well (if it isn't already).

**Comment by barbeau** [comment]
@caitbonnar Just curious - have you heard any feedback from B/LV users?

**Comment by caitbonnar** [comment]
@barbeau Not yet -- I sent an email out but I think they've been busy with the ASSETS conference in Portugal which was last week. I'll send out another note soon! Sorry about the delay. If it's holding you from releasing, I'd say to just include what you have now, it seems sound. :)

**Comment by alanborning** [comment]
BTW @meganca did move the StopInfo code to to the OneBusAway project at https://github.com/OneBusAway/onebusaway-stopinfo

**Comment by barbeau** [comment]
@caitbonnar No worries, thanks!  And thanks @alanborning, I did see that @meganca pushed the StopInfo code.

The StopInfo button on arrivals header in OBA is currently available in the `develop` branch (for regions that have a StopInfo server, which is currently only Puget Sound), so I'm going to call this issue closed, unless something comes up that makes the current implementation unusable, at which point we can re-open.  

FYI - I did add a fixed menu option for "Show Stop Details" (see https://github.com/OneBusAway/onebusaway-android/commit/3d502650459a83da0dc79e2818a6cb7000117f0d and #171), but for now that just shows the stop name, routes served, and stop code (which was the main item we needed to made visible somewhere).  After stop code is added to StopInfo page (see https://github.com/OneBusAway/onebusaway-stopinfo/issues/1), we can look at making this fixed menu option direct to the StopInfo page as well.
